### PR TITLE
Fruit list

### PR DESCRIPTION
**Pull request by carodriguezp**
please review

### CODE REVIEW

File: __mocks__/fileMock.js [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = "test-file-stub";
```
Comment (davepadrino) [inline __mocks__/fileMock.js:1]: Is this needed or is it a leftover?

File: src/App.style.ts [hunk 2]
```diff
@@ -1,8 +1,14 @@
 import styled from 'styled-components';
+import image from './pattern.png'
+import { Link } from "react-router-dom";
+import { ListGroupItem } from 'react-bootstrap';
+
+const fontColor = "#337AB7";
```
Comment (davepadrino) [inline src/App.style.ts:6]: Nice use of the variable!!

File: src/App.style.ts [hunk 3]
```diff
@@ -17,12 +23,76 @@ export const Container = styled.div`
...
+
+export const StyledLink = styled(Link)`
+text-decoration: none;
+&:hover {
+        
+        text-decoration: none;
+       
+
+}
+`
```
Comment (davepadrino) [inline src/App.style.ts:90]: Whitespace below

File: src/components/header/Header.tsx [hunk 4]
```diff
@@ -23,7 +23,7 @@ export default function Header() {
           </div>
           <div className="collapse navbar-collapse header-styled" id="myNavbar">
             <ul className="nav navbar-nav">
-              <li className="active">
+              <li className="active ">
```
Comment (davepadrino) [inline src/components/header/Header.tsx]: Remove extra space

File: src/components/main/fruits/fruitsListSection/__test__/FruitListSection.test.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import { mockFruitListFromApi } from "../../../../mocks";
+import FruitList from "../FruitList";
+import { render, screen } from "@testing-library/react";
+
+import "@testing-library/jest-dom";
+
+jest.mock("../../../../../services/api"); //route to be mocked
+
+describe("FruitList section calls API services", () => {
+  describe.only("when succesfull", () => {
```
Comment (davepadrino) [inline src/components/main/fruits/fruitsListSection/__test__/FruitListSection.test.tsx]: Remove `only` statement

File: src/components/main/fruits/fruitsListSection/__test__/FruitListSection.test.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    beforeEach(() => {
+      jest
+        .spyOn(myService, "getDataFromApi")
+        .mockResolvedValue(mockFruitListFromApi);
+    });
+
+    it("renders FruitList section", async () => {
+      render(<FruitList />);
+      //screen.debug();
```
Comment (davepadrino) [inline src/components/main/fruits/fruitsListSection/__test__/FruitListSection.test.tsx]: Remove commented line

File: src/components/main/fruits/fruitsListSection/__test__/FruitListSection.test.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      expect(list).toBeInTheDocument();
+    });
+  });
+
+  describe("when not succesfull", () => {
+    beforeEach(() => {
+      jest.spyOn(myService, "getDataFromApi").mockResolvedValue([]);
+    });
+
+    it("thows an error", async () => {
```
Comment (davepadrino) [inline src/components/main/fruits/fruitsListSection/__test__/FruitListSection.test.tsx]: Typo "throws"

File: src/components/main/fruits/fruitsListSection/fruitList.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+function FruitList() {
+  const [fruitsList, setFruitsList] = useState<APIResponse>([]);
+
+  const fetchFruits = async () => {
+    try {
+      const allFruits = await getDataFromApi();
+      setFruitsList(allFruits);
+    } catch (error) {
+      console.error("Sorry, there was an error.", error);
```
Comment (davepadrino) [inline src/components/main/fruits/fruitsListSection/fruitList.tsx:20]: Why you're not using the "error" param?

**Review by davepadrino — APPROVED** [review]
Small Changes, but overall looking good!

File: __mocks__/fileMock.js (hunk 1 above)
Comment (carodriguezp) [reply __mocks__/fileMock.js:1]: it is needed for testing purposes

**Pull request merged**